public/app/plugins/panel/pluginlist/PluginList.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import { GrafanaTheme, PanelProps, PluginMeta, PluginType } from '@grafana/data';
+import { config } from '@grafana/runtime';
+import { useAsync } from 'react-use';
+import { getBackendSrv } from 'app/core/services/backend_srv';
+import { UserDTO } from 'app/types';
+import { CustomScrollbar, ModalsController, stylesFactory, Tooltip, useStyles } from '@grafana/ui';
+import { UpdatePluginModal } from './components/UpdatePluginModal';
+import UserProvider from 'app/core/utils/UserProvider';
+import { css, cx } from 'emotion';
+
+export function PluginList(props: PanelProps) {
+  const pluginState = useAsync(async () => {
+    const plugins: PluginMeta[] = await getBackendSrv().get('api/plugins', { embedded: 0, core: 0 });
+    return [
+      { header: 'Installed Apps', list: plugins.filter((p) => p.type === PluginType.app), type: PluginType.app },
+      { header: 'Installed Panels', list: plugins.filter((p) => p.type === PluginType.panel), type: PluginType.panel },
+      {
+        header: 'Installed Datasources',
+        list: plugins.filter((p) => p.type === PluginType.datasource),
+        type: PluginType.datasource,
+      },
+    ];
+  }, []);
+
+  const styles = useStyles(getStyles);
+
+  if (pluginState.loading || pluginState.value === undefined) {
+    return null;
+  }
+
+  return (
+    <CustomScrollbar autoHeightMin="100%" autoHeightMax="100%">
+      <div className={styles.pluginList}>
+        {pluginState.value.map((category) => (
+          <div className={styles.section} key={`category-${category.type}`}>
+            <h6 className={styles.sectionHeader}>{category.header}</h6>
+            {category.list.map((plugin) => (
+              <a className={styles.item} href={plugin.defaultNavUrl} key={`plugin-${plugin.id}`}>
+                <img src={plugin.info.logos.small} className={styles.image} width="17" height="17" alt="" />
+                <span className={styles.title}>{plugin.name}</span>
+                <span className={styles.version}>v{plugin.info.version}</span>
+                <UserProvider userId={config.bootData.user.id}>
+                  {(api: any, states: any, teams: any, orgs: any, sessions: any, user?: UserDTO) => {
+                    if (!user?.isGrafanaAdmin) {
+                      return <></>;
```

Comment:
```suggestion
                      return null;
```
Any reason to return empty fragment here? It will be created as an object of type fragment, whereas returning null will skip that step.

Reply:
Guess it's because of the UserProvider typing, we could revert to Fragment when using contextSrv